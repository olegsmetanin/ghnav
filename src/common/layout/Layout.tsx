import * as React from 'react'

import { Theme, WithStyles, createStyles, withStyles } from '@material-ui/core/styles'

import Head from 'next/head'

const styles = (theme: Theme) =>
  createStyles({
    root: {
      paddingTop: theme.spacing.unit * 7
    }
  })

export interface ILayoutProps {
  title?: string
}

export const BaseLayout: React.SFC<ILayoutProps & WithStyles<typeof styles>> = ({
  children,
  classes,
  title = 'This is the default title'
}) => {
  return (
    <div className={classes.root}>
      <Head>
        <title>{title}</title>
      </Head>
      {children}
    </div>
  )
}

export const Layout = withStyles(styles)(BaseLayout)
