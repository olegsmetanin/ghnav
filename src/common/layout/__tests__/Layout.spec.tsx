import * as React from 'react'

/* eslint-env jest */
import { Layout } from '../Layout'
import renderer from 'react-test-renderer'

jest.mock('next/head', () => props => <div>{props.children}</div>)

describe('Layout', () => {
  it('renders correct', () => {
    const component = renderer.create(<Layout />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correct with title', () => {
    const component = renderer.create(<Layout title="Homepage">Content</Layout>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
