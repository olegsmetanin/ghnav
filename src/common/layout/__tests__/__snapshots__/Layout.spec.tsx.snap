// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correct 1`] = `
<div
  className="BaseLayout-root-1"
>
  <div>
    <title>
      This is the default title
    </title>
  </div>
</div>
`;

exports[`Layout renders correct with title 1`] = `
<div
  className="BaseLayout-root-1"
>
  <div>
    <title>
      Homepage
    </title>
  </div>
  Content
</div>
`;
